src/lib/Microsoft.Health.Events/EventHubProcessor/PartitionLockingProcessor.cs

```diff
@@ -0,0 +1,418 @@
+                        await Task.Delay(1000);
+                        continue;
+                    }
+
+                    throw;
+                }
+
+                _ownedPartitions.Add(partitionToClaim);
+                await Task.Delay(1000);
+            }
+        }
+
+        private static Pageable<BlobItem> ListBlobs(BlobContainerClient blobContainerClient, string prefix)
+        {
+            try
+            {
+                var resultSegment = blobContainerClient.GetBlobs(BlobTraits.Metadata, BlobStates.None, prefix);
+                return resultSegment;
+            }
+            catch (RequestFailedException e)
+            {
+                Console.WriteLine(e.Message);
+                Console.ReadLine();
+                throw;
+            }
+        }
+
+        private static async Task<int> GetActiveProcessors(Pageable<BlobItem> resultSegment, BlobContainerClient containerClient)
+        {
+            var activePods = new HashSet<string>();
+
+            foreach (BlobItem blobItem in resultSegment)
+            {
+                // if the processor has been updated in the last minute then consider it active
+                // otherwise delete it
+                if (blobItem.Metadata.TryGetValue("activeuntil", out var activeUntil))
+                {
+                    if (DateTime.Parse(activeUntil) > DateTime.UtcNow.AddMinutes(-1))
+                    {
+                        Console.WriteLine("Blob name: {0} is active", blobItem.Name);
+                        activePods.Add(blobItem.Name);
+                    }
+                    else
+                    {
+                        // todo: check the etag as a condition for delete
+                        Console.WriteLine($"Deleting blob {blobItem.Name}. It was last active {activeUntil}.");
+                        await containerClient.DeleteBlobAsync(blobItem.Name);
+                    }
+                }
+            }
+
+            return activePods.Count;
+        }
+
+        public async Task RunAndRenew(CancellationTokenSource cts)
+        {
+            // in the background we will continually update partition ownership until
+            // we receive a cancellation token to shutdown the processor
+
+            Console.WriteLine("Running the processor with partitions: ");
+            foreach (var p in _ownedPartitions)
+            {
+                Console.WriteLine(p);
+            }
+
+            try
+            {
+                await RenewOwnership(cts.Token);
+                await CheckProcessorCount(cts);
+                await StartStaticPartitionProcessor(cts.Token, _ownedPartitions);
+            }
+            catch (TaskCanceledException)
+            {
+                Console.WriteLine("A task was cancelled");
+            }
+        }
+
+        public async Task RunProcessor(CancellationTokenSource cts)
```

Comment:
The cancellation token should be wired through the stack and used with Task.Delay, etc.